packages/eas-cli/src/update/utils.ts

```diff
@@ -19,40 +21,46 @@ export type UpdateGroupDescription = FormatUpdateParameter & {
   runtimeVersion: string;
 };
 
+export type FormattedUpdateGroupDescription = {
+  message: string;
+  group: string;
+  platforms: string;
+  runtimeVersion: string;
+};
+
+export type FormattedUpdateGroupDescriptionWithBranch = FormattedUpdateGroupDescription & {
+  branch: string;
+};
+
 export const UPDATE_COLUMNS = [
   'Update message',
   'Update runtime version',
   'Update group ID',
   'Update platforms',
 ];
 
-export function getUpdateGroupsWithPlatforms<
-  UpdateFragment extends Pick<Update, 'platform' | 'group'>
->(updates: UpdateFragment[]): (UpdateFragment & { platforms: string })[] {
-  return Object.values(groupBy(updates, updates => updates.group)).map(updateGroup => ({
-    ...updateGroup[0],
-    platforms: formatPlatformForUpdateGroup(updateGroup),
-  }));
-}
+export const UPDATE_COLUMNS_WITH_BRANCH = ['Branch', ...UPDATE_COLUMNS];
 
 export function getPlatformsForGroup({
   group,
-  updates,
+  updates = [],
 }: {
-  group: string;
-  updates: { group: string; platform: string }[];
+  group: string | undefined;
+  updates: { group: string; platform: string }[] | undefined;
```

Comment:
once other comment is addressed, I think it makes the most sense to change the signature of this function to be `getPlatformsForGroup(updateGroup: { group: string; platform: string }[]): string` and then remove the groupBy and select (hoist those to callsites).

(we may not even need this function anymore and can just use `formatPlatformForUpdateGroup` which should probably be called `formatPlatformsStringForUpdateGroup`)

Reply:
This sounds great, but I'm worried about exploding this PR ins cope. This method is also used by `channel:view` and `branch:list` which work with the `UpdateBranchFragment` that fetches updates not update groups.

https://linear.app/expo/issue/ENG-6203/refactor-improve-updatebranchfragment-and-surrounding-code